3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/enrichments.scala

```diff
@@ -40,9 +37,7 @@ import utils.ScalazJson4sUtils
 /**
  * Trait inherited by every enrichment config case class
  */
-trait Enrichment {
-  val version: DefaultArtifactVersion
-}
+trait Enrichment
```

Comment:
So, as we deleted its single member, is there any other justification for its existence left?

Reply:
I think it is useful as a "marker trait" enabling things like the EnrichmentMap here: https://github.com/snowplow/snowplow/blob/362ebb527d4c392afc553615bd3c4a36230778b0/3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/package.scala#L65

but I may be wrong

Reply:
Yeah, it seems to enable some kind of reflection-based machinery, but I don't see why it cannot be implemented in less boilerplate and reflection-less way: https://github.com/snowplow/snowplow/blob/master/3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentRegistry.scala#L330. Do you think it is worth to create an issue to explore?

Reply:
Any opportunity to remove reflection is worth looking into. I created https://github.com/snowplow/snowplow/issues/3610